tools/serverpod_cli/lib/src/analyzer/models/checker/analyze_checker.dart

```diff
@@ -3,12 +3,6 @@ import 'package:serverpod_cli/src/analyzer/models/validation/keywords.dart';
 import 'package:yaml/yaml.dart';
 
 class AnalyzeChecker {
-  static bool isIdType(dynamic type) {
-    if (type is! String) return false;
-
-    return type == 'int' || type == 'int?';
-  }
-
```

Comment:
This was only used in one place. And there it was used like `AnalyzeChecker.isIdType(field.type.className)`. The better way to do that check is simply `field.type.isIdType`. Also, as `className` doesn't actually contain the `?` when called on a `TypeDefinition` of a nullable. I decided this was just better removed altogether.